Less warnings

grokys opened the pull request:
## What does the pull request do?

Ported a bunch of warning fixes from #4882 (thanks @workgroupengineering) and fixed the fixes that I felt were incorrect.

We still have 8 warnings that I'm not sure of the correct fix for.

grokys committed:
Add correct nullable for try pattern.

grokys committed:
Ported warning fixes from #4882.

Co-Authored-By: workgroupengineering <workgroupengineering@users.noreply.github.com>

kekekeks commented on src/Avalonia.Controls/ProgressBar.cs:
```diff
@@ -217,8 +217,10 @@ private void UpdateIndicator(Size bounds)
                     TemplateProperties.Container2AnimationEndPosition = barIndicatorWidth2 * 1.66; // Position at 166%
 
                     // Remove these properties when we switch to fluent as default and removed the old one.
```
I think we aren't removing the default theme. So we should probably somehow adapt it to the new properties instead.

grokys replied:
@maxkatz6 @danwalmsley

kekekeks commented on src/Avalonia.Controls/TextBox.cs:
```diff
@@ -542,7 +542,9 @@ public string RemoveInvalidCharacters(string text)
         }
 
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
         public async void Cut()
```
I guess we can just remove `async` here.

kekekeks commented on src/Avalonia.X11/X11Window.cs:
```diff
@@ -416,7 +415,9 @@ ev.ButtonEvent.button switch
                             8 => RawPointerEventType.XButton1Down,
-                            9 => RawPointerEventType.XButton2Down
+                            9 => RawPointerEventType.XButton2Down,
+                            _ => throw new AvaloniaInternalException(
```
I'd rather ignore the event completely than crash the app for no reason

grokys replied:
Ok, I didn't want to change the semantics (it would previously have thrown a `SwitchExpressionException` here) but that makes more sense to me.

grokys replied:
Actually, there's an `if` statement around the switch which should make sure that all cases are handled:

```
if (ev.ButtonEvent.button < 4 || ev.ButtonEvent.button == 8 || ev.ButtonEvent.button == 9)
```

Given that there's this check, and that this change isn't introducing an exception where there wouldn't be one before - I'd rather not refactor this logic in this PR. If you want to refactor it in a subsequent PR, I'd say that would be the best place for it.

kekekeks commented on src/Avalonia.X11/X11Window.cs:
```diff
@@ -444,7 +445,9 @@ ev.ButtonEvent.button switch
                             3 => RawPointerEventType.RightButtonUp,
                             8 => RawPointerEventType.XButton1Up,
-                            9 => RawPointerEventType.XButton2Up
+                            9 => RawPointerEventType.XButton2Up,
```
Same here

kekekeks commented on src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs:
```diff
@@ -171,9 +171,12 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
                                     RawPointerEventType.XButton1Down :
-                                    RawPointerEventType.XButton2Down
+                                    RawPointerEventType.XButton2Down,
+                                _ => throw new AvaloniaInternalException(
```
We should ignore the event instead of throwing from WndProc

grokys replied:
Here the `case`s in the switch statement must be the same as those in the enclosing `case`: hitting the `_` case means a logic error, so I think it makes sense to throw here.

grokys commented:
@kekekeks regarding the remaining C# warnings, it was stuff I've never really touched. It would be appreciated if you could take a quick look and push fixes to this branch, should only take 2 mins.

grokys committed:
Remove async from TextBox.Cut.

grokys committed:
Don't need to implement GetHashCode.

Instead use a `#pragma` to disable the warnings as we already do in `SelectorGrammar.cs` .

grokys committed:
Merge branch 'master' into fixes/less-warnings

grokys committed:
Merge branch 'master' into fixes/less-warnings

grokys committed:
Merge branch 'master' into fixes/less-warnings

workgroupengineering commented:
is there any chance this pr will be merged?

workgroupengineering commented:
I think this PR can be closed. 90% of the fixes have been applied.
